[core] Remove html option and make ClassImp* macros a no-op (#17543)
root-project/root: README/ReleaseNotes/v636/index.md

* The ROOT splash screen was removed for Linux and macOS
* Proof support has been completely removed form RooFit and RooStats, after it was already not working anymore for several releases
* The build options `mysql`, `odbc`, `pgsql` and `qt5web` have been deprecated. Please complain with root-dev@cern.ch should you still need one!
* The build option `html` has been removed.

> * The ClassImp macro and derived macros are now a no-op and will be deprecated in ROOT 6.38.

Reviewer: ```suggestion
* The ClassImp macro and derived macros have no effect now and will be deprecated in ROOT 6.38.
```
